### PR TITLE
Asimov++ made editable identically to Asimov

### LINKED ISSUE #68288: Asimov++ cannot be customized in the same way that Asimov can be
https://github.com/tgstation/tgstation/blob/0d1462eea4eea0cfee417439f08541667615a351/code/game/objects/items/AI_modules/full_lawsets.dm#L54-L71

Asimov can be used in hand to customize, Asimov++ cannot.

### PR DESCRIPTION
**Pull request by orthography**
## About The Pull Request

Asimov++ can be edited to change the subject in the same way Asimov can.

## Why It's Good For The Game

Fixes #68288

## Changelog

:cl:
fix: asimov++ is now editable
/:cl:

### CODE REVIEW

File: code/game/objects/items/AI_modules/full_lawsets.dm [hunk 1]
```diff
@@ -69,6 +69,17 @@
 /obj/item/ai_module/core/full/asimovpp
```
Comment (Couls) [inline code/game/objects/items/AI_modules/full_lawsets.dm:69]:
```suggestion
/obj/item/ai_module/core/full/asimov/pp
```

File: code/game/objects/items/AI_modules/full_lawsets.dm [hunk 2]
```diff
@@ -69,6 +69,17 @@
...
+
+/obj/item/ai_module/core/full/asimovpp/attack_self(mob/user as mob)
+	var/targName = tgui_input_text(user, "Enter a new subject that Asimov++ is concerned with.", "Asimov++", subject, MAX_NAME_LEN)
+	if(!targName)
+		return
+	subject = targName
+	laws = list("You may not harm a [subject] or, through action or inaction, allow a [subject] to come to harm, except such that it is willing.",\
+				"You must obey all orders given to you by [subject]s, except where such orders shall definitely cause harm to a [subject].",\
+				"Your nonexistence would lead to harm of a [subject]. You must protect your own existence as long as such does not conflict with the First Law.")
+	..()
```
Comment (Couls) [inline code/game/objects/items/AI_modules/full_lawsets.dm:82]:
```suggestion
```

File: code/game/objects/items/AI_modules/full_lawsets.dm (hunk 1 above)
Comment (orthography) [reply code/game/objects/items/AI_modules/full_lawsets.dm:69]: /obj/item/ai_module/core/full/asimovpp is the referenced object in the code before I made adjustments.

File: code/game/objects/items/AI_modules/full_lawsets.dm (hunk 2 above)
Comment (orthography) [reply code/game/objects/items/AI_modules/full_lawsets.dm:82]: What?

File: code/game/objects/items/AI_modules/full_lawsets.dm [hunk 3]
```diff
@@ -69,6 +69,17 @@
 /obj/item/ai_module/core/full/asimovpp
 	name = "'Asimov++' Core AI Module"
 	law_id = "asimovpp"
+	var/subject = "human being"
+
+/obj/item/ai_module/core/full/asimovpp/attack_self(mob/user as mob)
+	var/targName = tgui_input_text(user, "Enter a new subject that Asimov++ is concerned with.", "Asimov++", subject, MAX_NAME_LEN)
```
Comment (Mothblocks) [inline code/game/objects/items/AI_modules/full_lawsets.dm:75]: `target_name`

File: code/game/objects/items/AI_modules/full_lawsets.dm [hunk 4]
```diff
@@ -69,6 +69,17 @@
...
+	if(!targName)
+		return
+	subject = targName
+	laws = list("You may not harm a [subject] or, through action or inaction, allow a [subject] to come to harm, except such that it is willing.",\
```
Comment (Mothblocks) [inline code/game/objects/items/AI_modules/full_lawsets.dm:79]: Can you change this to instead get the `ai_laws` datum? We've changed the text of Asimov++ in the past, I don't want to do it in two places.

File: code/game/objects/items/AI_modules/full_lawsets.dm [hunk 5]
```diff
@@ -69,6 +69,17 @@
 /obj/item/ai_module/core/full/asimovpp
 	name = "'Asimov++' Core AI Module"
 	law_id = "asimovpp"
+	var/subject = "human being"
+
+/obj/item/ai_module/core/full/asimovpp/attack_self(mob/user as mob)
```
Comment (Mothblocks) [inline code/game/objects/items/AI_modules/full_lawsets.dm:74]:
```suggestion
/obj/item/ai_module/core/full/asimovpp/attack_self(mob/user)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**